gateway-server/src/main/java/org/apache/knox/gateway/services/topology/impl/DefaultTopologyService.java

```diff
@@ -153,17 +146,18 @@ private Topology loadTopology(File file) throws IOException, SAXException, Inter
     return topology;
   }
 
+  @Override
+  public Topology parse(final InputStream content) throws IOException, SAXException {
+    return TopologyUtils.parse(content);
+  }
+
   private Topology loadTopologyAttempt(File file) throws IOException, SAXException {
-    Topology topology;
-    Digester digester = digesterLoader.newDigester();
-    TopologyBuilder topologyBuilder = digester.parse(FileUtils.openInputStream(file));
-    if (null == topologyBuilder) {
-      return null;
-    }
-    topology = topologyBuilder.build();
-    topology.setUri(file.toURI());
-    topology.setName(FilenameUtils.removeExtension(file.getName()));
-    topology.setTimestamp(file.lastModified());
+    Topology topology = parse(FileUtils.openInputStream(file));
+    if (topology != null) {
+      topology.setUri(file.toURI());
+      topology.setName(FilenameUtils.removeExtension(file.getName()));
+      topology.setTimestamp(file.lastModified());
+    }
```

Comment:
If we pass in the file reference, you can move this logic in there too and get the full topology object back.